### PR TITLE
Feature basic styling main components

### PR DESCRIPTION
**Pull request by marcinel123**
Hi Adam,

In this PR I made following changes:
- created styles for Header and InputandFilterPanel components
- added basic stories files from storybook

Please have a look and let me know.
Martin

### CODE REVIEW

File: src/components/Header/Header.styles.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	display: flex;
+	justify-content: space-between;
+	align-items: center;
+	background-color: ${(props) => props.theme.colors.white};
+`;
+
+export const StyledH2 = styled.h2`
+	font-family: ${(props) => props.theme.fonts.family.basic};
+	font-size: ${(props) => props.theme.fonts.sizes.m};
+	@media screen and (min-width: 550px) {
```
Comment (AdamKniec) [inline src/components/Header/Header.styles.tsx]: 550px should be taken from some object with defined breakpoints

File: src/components/Header/Header.styles.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+		font-size: ${(props) => props.theme.fonts.sizes.l};
+	}
+`;
+
+export const StyledModeButton = styled.button`
+	font-family: ${(props) => props.theme.fonts.family.basic};
+	font-weight: ${(props) => props.theme.fonts.weight.semiBold};
+	font-size: ${(props) => props.theme.fonts.sizes.m};
+	padding: 10px 0px 10px 10px;
```
Comment (AdamKniec) [inline src/components/Header/Header.styles.tsx]: 0 is enough. No need to add px to 0

File: src/components/Header/Header.styles.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	border: none;
+	background-color: white;
```
Comment (AdamKniec) [inline src/components/Header/Header.styles.tsx]: take it from theme

File: src/components/Header/Header.styles.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	@media screen and (min-width: 550px) {
```
Comment (AdamKniec) [inline src/components/Header/Header.styles.tsx]: breakpoint should be defined in some config

File: src/components/Header/Header.styles.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+		font-size: ${(props) => props.theme.fonts.sizes.l};
+	}
+`;
+export const StyledButtonIcon = styled.i`
```
Comment (AdamKniec) [inline src/components/Header/Header.styles.tsx]: if it's a button - use a button tag

File: src/components/Header/Header.tsx [hunk 6]
```diff
@@ -1,6 +1,19 @@
...
-	<>
-		<h2>Where in the world?</h2>
-		<button type="button">Dark Mode</button>
-	</>
+	<StyledHeader>
+		<StyledH2>Where in the world?</StyledH2>
+		<StyledModeButton type="button">
+			<StyledButtonIcon
+				className="fa-regular fa-moon"
+				style={{ color: "#0d0d0d" }}
```
Comment (AdamKniec) [inline src/components/Header/Header.tsx]: remove inline styling

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
+import styled from "styled-components";
+import searchIcon from "../../images/search.svg";
+
+export const StyledInputandFilterSection = styled.section`
+	padding: 20px 40px;
+	width: 100%;
+	display: flex;
+	flex-direction: column;
+	align-items: start;
+	@media screen and (min-width: 550px) {
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: breakpoint should be defined in some config object

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+		flex-direction: row;
+		justify-content: space-between;
+		align-items: center;
+	}
+`;
+
+export const StyledForm = styled.form`
+	width: 100%;
+	@media screen and (min-width: 550px) {
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: the same as above - config object

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+`;
+
+export const StyledInputLabel = styled.label`
+	width: 100%;
+`;
+
+export const StyledInputElement = styled.input`
+	border: none;
+	width: 100%;
+	padding: 4% 0;
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: this seems weird. Try to get rid of such padding

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+	margin-bottom: 10%;
+
+	::placeholder {
+		background-image: url(${searchIcon});
+		background-repeat: no-repeat;
+		color: #b8b8b8;
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: this should be taken from theme

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+		padding-left: 5%;
+		@media screen and (min-width: 550px) {
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: config object 👍

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+			padding-left: 10%;
+		}
+	}
+	:focus {
+		::placeholder {
+			background-image: none;
+			color: white;
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: :)

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+		}
+	}
+	@media screen and (min-width: 550px) {
+		margin-bottom: 0;
+		width: 100%;
+		max-height: 58px;
+		padding: 4% 15px;
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: this looks off. Please remove percents here

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+`;
+export const StyledCountriesSelectLabel = styled.label`
+	width: 20%;
+	min-width: 145px;
+`;
+
+export const StyledCountriesSelect = styled.select`
+	border: none;
+	width: 100%;
+	padding: 7%;
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: this looks weird. Do not use % for paddings

File: src/stories/CountriesInfoPanel.stories.tsx [hunk 15]
```diff
@@ -0,0 +1,15 @@
+import { CountriesInfoPanel } from "../components/CountriesInfoPanel/CountriesInfoPanel";
+
+export default {
+	title: "CountriesInfo",
+	component: CountriesInfoPanel,
+};
+
+export const Default = () => <CountriesInfoPanel />;
+
+// const Template = (args) => <Template {...args} />;
```
Comment (AdamKniec) [inline src/stories/CountriesInfoPanel.stories.tsx]: remove this comment

File: src/stories/Header.stories.tsx [hunk 16]
```diff
@@ -0,0 +1,15 @@
+import { Header } from "../components/Header/Header";
+
+export default {
+	title: "Header",
+	component: Header,
+};
+
+export const Default = () => <Header />;
+
+// const Template = (args) => <Template {...args} />;
```
Comment (AdamKniec) [inline src/stories/Header.stories.tsx]: remove this comment

File: src/stories/InputandFilterPanel.stories.tsx [hunk 17]
```diff
@@ -0,0 +1,15 @@
+import { InputandFilterPanel } from "../components/InputandFilterPanel/InputandFilterPanel";
+
+export default {
+	title: "InputandFilterPanel",
+	component: InputandFilterPanel,
+};
+
+export const Default = () => <InputandFilterPanel />;
+
+// const Template = (args) => <Template {...args} />;
```
Comment (AdamKniec) [inline src/stories/InputandFilterPanel.stories.tsx]: same as above

**Comment by AdamKniec** [comment]
image
This is how app looks like on my machine. Am i missing something? ;)

File: src/components/Header/Header.styles.tsx (hunk 1 above)
Comment (marcinel123) [reply src/components/Header/Header.styles.tsx]: I have defined breakpoints in theme. Hope it is okay.

File: src/components/Header/Header.styles.tsx (hunk 2 above)
Comment (marcinel123) [reply src/components/Header/Header.styles.tsx]: fixed

File: src/components/Header/Header.styles.tsx (hunk 3 above)
Comment (marcinel123) [reply src/components/Header/Header.styles.tsx]: fixed

File: src/components/Header/Header.styles.tsx (hunk 4 above)
Comment (marcinel123) [reply src/components/Header/Header.styles.tsx]: I have defined breakpoints in theme. Hope it is okay.

File: src/components/Header/Header.styles.tsx (hunk 5 above)
Comment (marcinel123) [reply src/components/Header/Header.styles.tsx]: It is an icon inside a button. To be discussed together.

File: src/components/Header/Header.tsx (hunk 6 above)
Comment (marcinel123) [reply src/components/Header/Header.tsx]: fixed

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx (hunk 7 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: I have defined breakpoints in theme. Hope it is okay.

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx (hunk 8 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: I have defined breakpoints in theme. Hope it is okay.

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx (hunk 9 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: fixed

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx (hunk 10 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: fixed

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx (hunk 11 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: I have defined breakpoints in theme. Hope it is okay.

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx (hunk 12 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: fixed

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx (hunk 13 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: fixed

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx (hunk 14 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: fixed

File: src/stories/CountriesInfoPanel.stories.tsx (hunk 15 above)
Comment (marcinel123) [reply src/stories/CountriesInfoPanel.stories.tsx]: fixed

File: src/stories/Header.stories.tsx (hunk 16 above)
Comment (marcinel123) [reply src/stories/Header.stories.tsx]: fixed

File: src/stories/InputandFilterPanel.stories.tsx (hunk 17 above)
Comment (marcinel123) [reply src/stories/InputandFilterPanel.stories.tsx]: fixed

File: src/components/CountriesInfoPanel/CountriesInfoPanel.tsx [hunk 18]
```diff
@@ -1 +1,5 @@
-export const CountriesInfoPanel = () => <div>CountriesInfoPanel</div>;
+import { StyledDisplaySection } from "./CountriesInfoPanel.styles";
+
+export const CountriesInfoPanel = () => (
```
Comment (AdamKniec) [inline src/components/CountriesInfoPanel/CountriesInfoPanel.tsx]: Please rename to CountriesList

File: src/components/Header/Header.styles.tsx [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+
+export const StyledH2 = styled.h2`
+	font-family: ${(props) => props.theme.fonts.family.basic};
+	font-size: ${(props) => props.theme.fonts.sizes.m};
+	@media screen and (min-width: ${(props) => props.theme.breakpoints.medium}) {
+		font-size: ${(props) => props.theme.fonts.sizes.l};
+	}
+`;
+
+export const StyledModeButton = styled.button`
```
Comment (AdamKniec) [inline src/components/Header/Header.styles.tsx]: Change name to ThemeSwitchButton

File: src/components/Header/Header.styles.tsx (hunk 5 above)
Comment (AdamKniec) [reply src/components/Header/Header.styles.tsx]: Change name to StyledIcon

File: src/components/CountriesInfoPanel/CountriesInfoPanel.tsx (hunk 18 above)
Comment (marcinel123) [reply src/components/CountriesInfoPanel/CountriesInfoPanel.tsx]: fixed

Comment (marcinel123) [reply src/components/CountriesInfoPanel/CountriesInfoPanel.tsx]: fixed

File: src/components/Header/Header.styles.tsx (hunk 19 above)
Comment (marcinel123) [reply src/components/Header/Header.styles.tsx]: fixed

**Review by AdamKniec — APPROVED** [review]
🎖️

**Pull request merged**